### ISSUE TITLE
Support Node.js scoping as an option?

### ISSUE THREAD

**Issue by nzakas** [estools/escope#43]
Would you consider supporting Node.js scoping? Node.js does a weird thing where the module code is actually run inside a function, so this:

```js
return false;
```

is legal even though it's not in a function because it's actually interpreted as:

```js
(function(require, module, process) {
return false;
}());
```

What I'm proposing is a new flag `nodejsScope` that can be set to `true` to artificially add a function scope immediately following the global scope. 

That way, tools like ESLint can properly maintain scope even when evaluating Node.js files.

**Comment by Constellation** [comment]
This suggestion looks reasonable and nice.
Do you have a patch? Or is it preferable that I create the patch?

**Comment by nzakas** [comment]
If you have the time, please go ahead. :) If not, I'll probably have time in the next couple of weeks to attempt a patch.

**Comment by Constellation** [comment]
OK, since now I have some time, I'll implement it ;)

**Comment by Constellation** [comment]
OK, I've released it as `2.0.6` (backporting)

**Comment by nzakas** [comment]
Thanks!
